tensorflow/core/kernels/cwise_ops.h

```diff
@@ -401,6 +401,32 @@ struct floor : base<T, Eigen::internal::scalar_floor_op<T>> {};
 template <typename T>
 struct ceil : base<T, Eigen::internal::scalar_ceil_op<T>> {};
 
+namespace impl {
+#ifndef __CUDACC__
+template <typename T >
+T rint(T v){
+    return std::rint(v);
+}
+#else
+template <typename T>
+EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE T rint(T v) {
+    return ::rint(v);
+}
+#endif
+} //end namespace impl
+
+/** this should go in Eigen
+  * \brief Template functor to compute the round to int value of a scalar
+  */
+template<typename Scalar> struct scalar_rint_op {
+  EIGEN_EMPTY_STRUCT_CTOR(scalar_rint_op)
+  EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE const Scalar operator() (const Scalar& a) const { return impl::rint(a); }
```

Comment:
80 columns.  Also, there doesn't seem to be any reason to declare the auxiliary `impl::rint` function.  Can't you just move all the `#ifdef`s into this `operator()`?